### PR TITLE
Save selected player in database

### PR DESCRIPTION
**Pull request by Susanne1307**
When choosing a player, the player (name and image source) will be saved within the logged in user.

### CODE REVIEW

File: client/src/components/PlayerDropdown.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default function PlayerDropdown({ value, onChange }) {
 
   return (
     <Dropdown value={value ? value.name : ''} onChange={handleChange}>
-      <Option disabled value=""></Option>
+      <Option defaultValue>What&apos;s your name?</Option>
```
Comment (lmachens) [inline client/src/components/PlayerDropdown.js:34]: `defaultValue` is empty

File: client/src/contexts/UserProvider.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import Logout from '../pages/Logout';
+
+const UserProvider = (props) => {
+  const token = localStorage.getItem('loggedInUserToken');
+  try {
+    let base64Url = token.split('.')[1];
+    let base64 = base64Url.replace('-', '+').replace('_', '/');
+    let decodedData = JSON.parse(
+      Buffer.from(base64, 'base64').toString('binary')
+    );
```
Comment (lmachens) [inline client/src/contexts/UserProvider.js:13]:
```suggestion
    const base64Url = token.split('.')[1];
    const base64 = base64Url.replace('-', '+').replace('_', '/');
    const decodedData = JSON.parse(
      Buffer.from(base64, 'base64').toString('binary')
    );
```

File: client/src/contexts/UserProvider.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    const loggedInUserId = decodedData.userId;
+
+    return (
+      <UserContext.Provider value={[loggedInUserId]}>
```
Comment (lmachens) [inline client/src/contexts/UserProvider.js:17]:
```suggestion
      <UserContext.Provider value={loggedInUserId}>
```

File: client/src/pages/Profile.js [hunk 4]
```diff
@@ -22,34 +24,40 @@ const Image = styled(ProfileImage)`
...
     setPlayer(player);
-  };
-  const playerImageSrc = player ? player.img : example;
+  }
 
   return (
     <>
       <FullContainer>
         <Header />
+        {loading}
```
Comment (lmachens) [inline client/src/pages/Profile.js:56]: `loading` is a boolean. Nothing will be rendered here.

File: lib/routes/auth.js [hunk 5]
```diff
@@ -45,4 +46,33 @@ router.post('/login', async (request, response) => {
...
+});
+
+router.get('/:id', async (request, response) => {
+  const userId = request.params.id;
+  try {
+    const user = await User.findById(userId);
+    response.send(user);
+  } catch (error) {
+    throw new Error(error.message);
+  }
```
Comment (lmachens) [inline lib/routes/auth.js:62]: First, you are using `.then` and `.catch`, then it's `try-catch`. I wouldn't mix it.

File: client/src/contexts/UserProvider.js (hunk 3 above)
Comment (Susanne1307) [reply client/src/contexts/UserProvider.js:17]: If I leave out the brackets, the patch will no longer work. It seems to take only the first digit of the Id?
`users.js:59 PATCH http://localhost:3000/api/users/5 500 (Internal Server Error)`

**Pull request merged**